code_annotations/extensions/base.py

```diff
@@ -19,14 +26,219 @@ def __init__(self, config, echo):
+
+    r"""
+    This format string/regex finds our annotation token and choices / comments inside a comment:
+
+    [\s\S]*? - Strip out any characters between the start of the comment and the annotation
+    ({})     - {} is a Python format string that will be replaced with a regex escaped and
+               then or-joined to make a list of the annotation tokens we're looking for
+               Ex: (\.\.\ pii\:\:|\.\.\ pii\_types\:\:)
+    (.*)     - and capture all characters until the end of the line
+
+    Returns a 2-tuple of found annotation token and annotation comment
+
+    TODO: Make multi line annotation comments work again.
+    """
+    annotation_regex = r'[\s\S]*?({})(.*)'
+
+    def __init__(self, config, echo):
+        """
+        Set up the extension and create the regexes used to do searches.
+
+        Args:
+            config: The configuration dict
+            echo: VerboseEcho object for logging
+        """
+        # These must be set before the parent config is called since callbacks happen
+        # on parent __init__ to _add_annotation_token / _add_annotation_group.
+        self.annotation_tokens = []
+        self.annotation_regexes = []
+
+        super(SimpleRegexAnnotationExtension, self).__init__(config, echo)
+
+        if self.lang_comment_definition is None:  # pragma: no cover
+            raise ValueError('Subclasses of SimpleRegexAnnotationExtension must define lang_comment_definition!')
+
+        # pylint: disable=not-a-mapping
+        self.comment_regex = self.comment_regex_fmt.format(**self.lang_comment_definition)
+
+        # Parent class will allow this class to populate self.strings_to_search via
+        # calls to _add_annotation_token or _add_annotation_group for each configured
+        # annotation.
+        self.query = self.annotation_regex.format('|'.join(self.annotation_regexes))
+
+        self.ECHO.echo_v("{} extension regex query: {}".format(self.extension_name, self.query))
+
+    def _add_annotation_token(self, annotation):
+        """
+        Add annotations to our local lists during configuration.
+
+        Args:
+            annotation: An annotation token (ex. ".. pii::")
+
+        Raises:
+            TypeError
+        """
+        self.annotation_tokens.append(annotation)
+        self.annotation_regexes.append(re.escape(annotation))
+
+    def _add_annotation_group(self, annotation_group):
+        """
+        Add an annotation group to our local lists during configuration.
+
+        Args:
+            annotation_group: An annotation group from the configuration file.
+
+        Raises:
+            TypeError
+        """
+        for annotation in annotation_group:
+            if isinstance(annotation, six.string_types):
+                self._add_annotation_token(annotation)
+            elif isinstance(annotation, dict):
+                annotation_name = next(iter(annotation))
+                self._add_annotation_token(annotation_name)
+            else:
```

Comment:
Is there intentionally no `elif` for lists/tuples here? The exception below indicates they should be handled as well.

Reply:
Opposite problem, that was copypasta. I'll update the exception.